src/main/java/com/nilsenlabs/flavormatrix/actions/AndroidModuleHelper.kt

```diff
@@ -24,50 +23,118 @@ object AndroidModuleHelper {
     }
 
     /**
-     * Fills the list with a merged set of dimensions for all modules
+     * Fills the list with a merged set of dimensions for all models
+     * Uses reflection-based access to support multiple IDE versions
      */
-    private fun createMergedDimensionList(modules: List<GradleAndroidModel>): DimensionList {
+    private fun createMergedDimensionList(models: List<Any?>): DimensionList {
         val dimensionList = DimensionList()
-        for (module in modules) {
-            val flavors = module.androidProject.multiVariantData?.productFlavors?.toList() ?: continue
-            for (flavorObj in flavors) {
-                // getLog.info("Processing flavorName: ${flavorObj.productFlavor.name}: ${flavorObj.productFlavor.dimension}")
-                /*
-                    Expecting a log like like this for my test project:
-                    Processing flavorName: qa: environment
-                    Processing flavorName: prod: environment
-                    Processing flavorName: dev: environment
-                    Processing flavorName: paid: paymentmodel
-                    Processing flavorName: free: paymentmodel
-                 */
-                val flavor = flavorObj.productFlavor
-                flavor.dimension?.let { dim ->
-                    val flavorsForDimension = dimensionList.getOrCreateDimension(dim)
-                    flavorsForDimension.addUniqueVariant(flavor.name)
+        
+        getLog().debug("Starting dimension analysis for ${models.size} models...")
+        
+        for ((index, model) in models.withIndex()) {
+            try {
+                getLog().debug("Processing model $index: ${model?.javaClass?.name}")
+                
+                // Try the dependency model approach first for better compatibility
+                if (model?.javaClass?.name?.contains("GradleAndroidDependencyModel") == true) {
+                    getLog().debug("Attempting to extract dimensions from dependency model...")
+                    
+                    // Try to get flavor dimensions directly from the dependency model
+                    try {
```

Comment:
The cycolmatic complexity of this function is quite high. Is line 43 maybe a point where you could extract the block to a separate function? (I'm unsure of what to name it - haven't had time to dig into what it does)